manifests/allow.pp

```diff
@@ -2,19 +2,19 @@
 #
 # @param name [String]
 #
+# @param rules
+#   A standard ``ntpd.conf`` restrict rule (``notrust``, etc...)
+#
 # @param trusted_nets
 #   Networks and Hosts to allow
 #
-# @param rules
-#   A standard ``ntpd.conf`` restrict append rule (``notrust``, etc...)
-#
 # @param firewall
 #   If enabled, allow connections from `trusted_nets`
 #
 define ntpd::allow (
-  Simplib::Netlist $trusted_nets = simplib::lookup('simp_options::trusted_nets', { 'default_value' => ['127.0.0.1', '::1'] }),
-  Optional[String] $rules        = undef,
-  Boolean          $firewall     = simplib::lookup('simp_options::firewall', { 'default_value' => false})
+  Optional[String[1]] $rules        = undef,
```

Comment:
If we make this accept a string or an array of the new Ntpd::Restrict, this requires no change to the existing template, is a backward-compatible interface, but can be changed to simply the array at the later date.  The array of Ntpd::Restrict provides validation for the user.